res/schema.xml

```diff
@@ -480,5 +480,13 @@ METADATA
         position INTEGER
         );
     </sql>
-  </revision>  
+  </revision>
+  <revision version="31" min_compatible="3">
+    <description>
+      Add track color support. Default color is #000000.
+    </description>
+    <sql>
+      ALTER TABLE library ADD COLUMN color INTEGER DEFAULT 0xFF000000 NOT NULL;
```

Comment:
- The track color column should be nullable. Let's not repeat the mistake that happened with the cue colors. This cannot be changed afterwards without cloning and rebuilding the whole table.
- The column should be initialized with NULL as the default value
- The values should strictly be 0xRRGGBB, i.e. no alpha channel

Reply:
Alternatively we could use 100% transparency to represent null. From a UX perspective, I don't think the difference matters. As for the code, I don't know if one would be preferable to the other.

Reply:
Don't mess around with the database schema, just be explicit. Use RGB + NULL in the database and a thin `OptionalRgbColor` adapter around QColor that handles all valid RGB values (0xFFRRGGBB) and the special NULL value (0x00000000) with various internal consistency checks. Just as I once proposed.

Reply:
I will provide this class as a starting point, just to demonstrate its usefulness. Stay tuned.

Reply:
This [small class](#2472) handles our use cases:

- Integer RGB color codes in the database and UI layer
- QColor in the application layer
- Explicit *no color* state

It reduces the complexity of directly dealing with `QColor`and is essentially just a `QColor`.

q.o.d.